lib/event.rb

```diff
@@ -0,0 +1,41 @@
+class Event
+  attr_reader :name,
+              :crafts,
+              :attendees
+  def initialize(name, crafts, attendees)
+    @name = name
+    @crafts = []
+    @attendees = []
+  end
+
+  def add_crafts(craft)
+    @crafts << craft
+  end
+
+  def add_attendees(person)
+    @attendees << person
+  end
+
+  def attendee_names
+    @attendees.map do |attendee|
+      attendee.name
+    end
+  end
+
+  def craft_with_most_supplies
+    craft = @crafts.max_by do |craft|
+      craft.supplies_required.length
+    end
+    craft.name
+  end
+
+  def supply_list
+    supplies = []
```

Comment:
This one didn't pass the harness either but when I tried it in mine it worked. I am going to give you credit for it anyway. Here is what I came up with 
```
def supply_list
    supply_array = @crafts.map do |craft|
      craft.supplies_required.keys
    end
    names_as_string = supply_array.flatten.map do |supply_name|
      supply_name.to_s
    end
    names_as_string.uniq
  end
```